src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordDeconstructor.cs

```diff
@@ -0,0 +1,158 @@
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations => TypeWithAnnotations.Create(
+            isNullableEnabled: true,
+            ContainingType.DeclaringCompilation.GetSpecialType(SpecialType.System_Void));
+
+        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => FlowAnalysisAnnotations.None;
+
+        public override ImmutableHashSet<string> ReturnNotNullIfParameterNotNull => ImmutableHashSet<string>.Empty;
+
+        public override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotations
+            => ImmutableArray<TypeWithAnnotations>.Empty;
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations => ImmutableArray<MethodSymbol>.Empty;
+
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
+
+        public override Symbol? AssociatedSymbol => null;
+
+        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
+
+        internal override bool GenerateDebugInfo => false;
+
+        public override Symbol ContainingSymbol => ContainingType;
+
+        public override ImmutableArray<Location> Locations => ContainingType.Locations;
+
+        public override Accessibility DeclaredAccessibility => Accessibility.Public;
+
+        public override bool IsStatic => false;
+
+        public override bool IsVirtual => false;
+
+        public override bool IsOverride => false;
+
+        public override bool IsAbstract => false;
+
+        public override bool IsSealed => false;
+
+        public override bool IsExtern => false;
+
+        internal override LexicalSortKey GetLexicalSortKey()
+        {
+            // We need a separate sort key because struct records will have two synthesized
+            // constructors: the record constructor, and the parameterless constructor
+            return LexicalSortKey.SynthesizedRecordDeconstructor;
+        }
+
+        internal override bool SynthesizesLoweredBoundBody => true;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var F = new SyntheticBoundNodeFactory(this, ContainingType.GetNonNullSyntaxNode(), compilationState, diagnostics);
+
+            Debug.Assert(Parameters.Length == _members.Length);
+            var statementsBuilder = ArrayBuilder<BoundStatement>.GetInstance(_members.Length + 1);
+            for (int i = 0; i < _members.Length; i++)
+            {
+                var parameter = Parameters[i];
+                var member = _members[i];
+                var propertyAccess = member switch
+                {
+                    PropertySymbol prop => F.Property(F.This(), prop),
+                    FieldSymbol field => F.Field(F.This(), field),
```

Comment:
>FieldSymbol field => F.Field(F.This(), field), [](start = 20, length = 46)

We should not be able to get here with anything but a property (because a property is required to exist for each record parameter).

Reply:
Now limiting the allowable symbol kinds to properties.